scylla/src/transport/connection.rs

```diff
@@ -217,6 +217,8 @@ pub(crate) struct QueryResponse {
     pub(crate) response: Response,
     pub(crate) tracing_id: Option<Uuid>,
     pub(crate) warnings: Vec<String>,
+    #[allow(dead_code)] // This is not exposed to user (yet?)
+    pub(crate) custom_payload: Option<HashMap<String, Vec<u8>>>,
```

Comment:
It looks like it would still require more changes in scylla-cql than I'm comfortable with in this PR.
custom_payload is rarely used feature (pre-tablets), so we can assume that without tablets those allocations won't happen.
With tablets it's 1 additional allocation. I think we can live with this for now.

I'll open an issue so we can improve that in the future.

Reply:
Opened: https://github.com/scylladb/scylla-rust-driver/issues/984